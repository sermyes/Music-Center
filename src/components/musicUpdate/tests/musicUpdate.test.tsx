import MusicUpdate from '../musicUpdate';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { fakeVideos } from '../../tests/videos';
import userEvent from '@testing-library/user-event';

describe('Music Update', () => {
  const videos = fakeVideos;
  const onVideoClick = jest.fn();

  it('renders correctly', () => {
    const component = renderer.create(
      <MusicUpdate updatedVideos={videos} onVideoClick={onVideoClick} />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('calls onVideoClick with video when clicked', async () => {
    render(<MusicUpdate updatedVideos={videos} onVideoClick={onVideoClick} />);
    const videoItem = screen.getByText('title2');
    await userEvent.click(videoItem);

    expect(onVideoClick).toHaveBeenCalledTimes(1);
    expect(onVideoClick).toHaveBeenCalledWith(videos[1]);
  });
});
