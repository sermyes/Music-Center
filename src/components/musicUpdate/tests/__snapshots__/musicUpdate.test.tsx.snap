// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Music Update renders correctly 1`] = `
Array [
  <div
    className="titleContainer"
  >
    <h2
      className="title"
    >
      Updated Music
    </h2>
  </div>,
  <div
    className="listContainer"
  >
    <div
      className="list"
    >
      <div
        className="container"
      >
        <div
          className="video"
          onClick={[Function]}
          title="video"
        >
          <img
            alt="title1"
            className="thumbnail"
            src="https://image/"
          />
          <div
            className="metadata"
          >
            <p
              className="title"
            >
              title1
            </p>
          </div>
        </div>
      </div>
      <div
        className="container"
      >
        <div
          className="video"
          onClick={[Function]}
          title="video"
        >
          <img
            alt="title2"
            className="thumbnail"
            src="https://image/"
          />
          <div
            className="metadata"
          >
            <p
              className="title"
            >
              title2
            </p>
          </div>
        </div>
      </div>
      <div
        className="container"
      >
        <div
          className="video"
          onClick={[Function]}
          title="video"
        >
          <img
            alt="title3"
            className="thumbnail"
            src="https://image/"
          />
          <div
            className="metadata"
          >
            <p
              className="title"
            >
              title3
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
